### PR TITLE
feat(router): add tests for router

### PR DESCRIPTION
**Pull request by walktall**
**What this PR does**

- Add more tests for router.

- Refine some code when adding tests.

**why we need it**

Fix https://github.com/caicloud/nirvana/issues/13

**Special notes for your reviewer**:

@kdada @ddysher

### CODE REVIEW

**Comment by walktall** [comment]
@kdada
 
`Parse("localhost")`
`Parse("api/v1")`

Should return an error.

File: router/router.go [hunk 1]
```diff
@@ -289,9 +289,9 @@ type ExpSegment struct {
 	Key string
 }
 
-// ParseExpSegment parses a regexp segment to ExpSegment.
-func ParseExpSegment(exp string) (*ExpSegment, error) {
-	if !strings.HasPrefix(exp, "{") || !strings.HasSuffix(exp, "}") {
+// parseExpSegment parses a regexp segment to ExpSegment.
+func parseExpSegment(exp string) (*ExpSegment, error) {
```
Comment (ddysher) [inline router/router.go]: Please use a different PR for this

File: router/router.go [hunk 2]
```diff
@@ -289,9 +289,9 @@ type ExpSegment struct {
...
+	if !strings.HasSuffix(exp, "}") || !strings.HasPrefix(exp, "{") {
```
Comment (ddysher) [inline router/router.go]: why changing the order?

Comment (walktall) [reply router/router.go]: The first `{` checking was done before the execution going into this func. So it was redundant.
In fact, we only need to check `}`.

File: router/router.go (hunk 1 above)
Comment (walktall) [reply router/router.go]: Then it is better to leave it to @kdada I think.

**Comment by walktall** [comment]
@ddysher Got 1 bug. `localhost` and `api/v1` will pass the examination. Already @kdada in comment.

**Comment by kdada** [comment]
@walktall 

Why do you consider it's a bug?

**Comment by walktall** [comment]
@kdada 
`Parse` didn't report error, but `Match` failed.

**Comment by Jimexist** [comment]
is this still WIP?

**Comment by walktall** [comment]
@Jimexist Yes.

**Comment by ddysher** [comment]
Need to update after conclusion from https://github.com/caicloud/nirvana/issues/24.

**Comment by walktall** [comment]
@ddysher @Jimexist PTAL

**Comment by Jimexist** [comment]
@walktall tests failed

**Comment by walktall** [comment]
ping @Jimexist

**Comment by Jimexist** [comment]
merge to move forward.

**Pull request merged**